plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -6183,6 +6183,8 @@ public Type getType() {
                         dsMo.setCustomFieldValue(CustomFieldConstants.CLOUD_UUID, poolUuid);
                     }
 
```

Comment:
Will this create fcd folder for all storage types or just for vmfs? I'm asking as another solidfire related PR does the same for solidfire: https://github.com/apache/cloudstack/pull/4977/files

Reply:
Currently, we are creating this folder in two places, 1. modifystoragepool command and 2. initialize local storage.
I think solidfire is not covered in these two places.
We need to see the right place for solidfire. The folder creation in PR #4977 may work but it is a kind of hack to check the folder existence upon an operation, ideally, it has to create when MS initialises the storage. Let me know if you know such a place otherwise, I'll incorporate the changes made in #4977

Reply:
I've tested the fix in PR #4977 but it did not work. The place where it fixed is not the common code path to all workflows. I'm working on another fix.

Reply:
ModifyStoragePool command is not addressed for solidfire use case due to which this is failing there. Maybe they can handle this as part of PR #4977. For local storage, we can continue with this PR.